### PR TITLE
[#21] [Frontend] As a user, I can see the detail of a TIL

### PR DESCRIPTION
**Pull request by tyrro**
Closes https://github.com/nimblehq/til-web/issues/21

## What happened 👀

After implementing the backend in https://github.com/nimblehq/til-web/issues/12, this PR implements the frontend!

## Insight 📝

- `PostCard` is modified to reduce the number of nested divs
- `content` attribute is moved into Post basic fields, `ogImage` is added to the extended list of fields
- `PostDetails` component is created to show post details including `tags`

## Proof Of Work 📹

image

### CODE REVIEW

**Review by Lahphim — APPROVED** [review]

File: src/pages/posts/[slug].page.tsx [hunk 1]
```diff
@@ -36,10 +38,11 @@ const PostDetails = ({ post }: PostProps) => {
 
       <h1
         className="m-8 text-7xl items-center text-center"
-        data-test-id={postDetailsTestIds.title}
+        data-test-id={postDetailsTestIds.homePageLink}
       >
-        {post.title}
+        <Link href="/">TIL</Link>
```
Comment (hoangmirs) [inline src/pages/posts/[slug].page.tsx:43]: Should h1 be the title of the post instead? 
And the home page link could be h2 🤔

Comment (tyrro) [reply src/pages/posts/[slug].page.tsx:37]: As we are using `text-*` classes, I resized them in 8079f722b4ac21f0a058ce1a981817c8191b3f98, Thank you!

File: src/lib/post.ts [hunk 2]
```diff
@@ -38,9 +38,10 @@ const BASIC_FIELDS = [
   'author',
   'coverImage',
   'excerpt',
+  'content',
 ] as Field[];
 
-const EXTENDED_FIELDS = [...BASIC_FIELDS, 'content', 'tags'] as Field[];
+const EXTENDED_FIELDS = [...BASIC_FIELDS, 'ogImage', 'tags'] as Field[];
```
[the remark this answers is not in the document]
Comment (tyrro) [reply src/lib/post.ts]: good idea, I think we don't need any distinctions between basic and extended fields for now as both differ very little, merged them into a single `POST_FIELDS` in 2349a202c3c5c9b06219f1b887ad059bc743c836, Thank you!

File: src/pages/posts/[slug].page.tsx (hunk 1 above)
Comment (hoangmirs) [reply src/pages/posts/[slug].page.tsx:37]: Should the `TIL` text on the Post Details page be the same as `TIL` on the home page?

**Review by hoangmirs — APPROVED** [review]

**Pull request merged**